pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentGenerationTaskRunner.java

```diff
@@ -167,19 +167,25 @@ private SegmentNameGenerator getSegmentNameGenerator(SegmentGeneratorConfig segm
         return new InputFileSegmentNameGenerator(segmentNameGeneratorConfigs.get(FILE_PATH_PATTERN),
             segmentNameGeneratorConfigs.get(SEGMENT_NAME_TEMPLATE), inputFileUri, appendUUIDToSegmentName);
       case BatchConfigProperties.SegmentNameGeneratorType.UPLOADED_REALTIME:
-        Preconditions.checkState(segmentGeneratorConfig.getCreationTime() != null,
```

Comment:
uploadTime is a property of the ingestionJob to provide deterministic names to segments based on when the job ran and re-running replaces the segments. Moving this to segmentNameGeneratorConfig means it is no more a job property and can be set independently.

The upstream job should have the responsibility to either use its creation time or default to currentTime/any other time.

Reply:
If you look at the original issue: https://github.com/apache/pinot/issues/14083
The reporter got an error `Creation time must be set for uploaded realtime segment name generator`. I think this means it's not a default property of the ingestion job, otherwise he shouldn't get this error. So I moved it to segmentNameGeneratorConfig and simply use current system time in the upstream job

Reply:
Can we use `segmentNameGeneratorConfigs.get(BatchConfigProperties.SEGMENT_UPLOAD_TIME_MS)` and also fall back to `segmentGeneratorConfig.getCreationTime() for ` [UploadedRealtimeSegmentName](https://github.com/apache/pinot/blob/07962b0adefe2af8c5698d8d674f35a395d7b129/pinot-common/src/main/java/org/apache/pinot/common/utils/UploadedRealtimeSegmentName.java#L42)

I feel this multiple places for providing create time can be confusing.